src/test/java/org/breedinginsight/api/auth/rules/ProgramSecuredAnnotationRuleIntegrationTest.java

```diff
@@ -0,0 +1,214 @@
+                .name("Another test program")
+                .species(speciesRequest)
+                .build();
+
+        Flowable<HttpResponse<String>> call = client.exchange(
+                POST("/programs", gson.toJson(validRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .cookie(new NettyCookie("phylo-token", "test-registered-user")), String.class
+        );
+
+        HttpResponse<String> response = call.blockingFirst();
+        assertEquals(HttpStatus.OK, response.getStatus());
+    }
+
+    @Test
+    public void programRoleGroupMatch() {
+        // User has role required in program role group
+        Flowable<HttpResponse<String>> call = client.exchange(
+                GET("/programs/" + programs.get(0).getId().toString())
+                        .cookie(new NettyCookie("phylo-token", "other-registered-user")), String.class
+        );
+
+        HttpResponse<String> response = call.blockingFirst();
+        assertEquals(HttpStatus.OK, response.getStatus());
+
+        JsonObject result = JsonParser.parseString(response.body()).getAsJsonObject().getAsJsonObject("result");
+
+        assertEquals(programs.get(0).getId().toString(), result.get("id").getAsString(), "Response not returned");
+    }
+
+    @Test
+    public void programRoleMissing() {
+        // User missing role required
+        SpeciesRequest speciesRequest = SpeciesRequest.builder()
+                .id(programs.get(0).getSpecies().getId())
+                .build();
+
+        ProgramRequest validRequest = ProgramRequest.builder()
+                .name("Another test program")
+                .species(speciesRequest)
+                .build();
+
+        Flowable<HttpResponse<String>> call = client.exchange(
+                POST("/programs", gson.toJson(validRequest))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .cookie(new NettyCookie("phylo-token", "other-registered-user")), String.class
+        );
+
+        HttpClientResponseException e = Assertions.assertThrows(HttpClientResponseException.class, () -> {
+            HttpResponse<String> response = call.blockingFirst();
+        });
+
+        assertEquals(HttpStatus.FORBIDDEN, e.getStatus());
+    }
+
+    @Test
+    public void programRoleGroupMissing() {
+        // User missing role required in program role group
+        Flowable<HttpResponse<String>> call = client.exchange(
+                GET("/programs/" + programs.get(3).getId().toString())
+                        .cookie(new NettyCookie("phylo-token", "other-registered-user")), String.class
+        );
+
+        HttpClientResponseException e = Assertions.assertThrows(HttpClientResponseException.class, () -> {
+            HttpResponse<String> response = call.blockingFirst();
+        });
+
+        assertEquals(HttpStatus.FORBIDDEN, e.getStatus());
+    }
+
+    @Test
+    @Order(12)
```

Comment:
interesting (talking about the @Order(12) haha)

Reply:
Hahah copy and paste strikes again! Changed and pushed